webapp/src/actions/index.js

```diff
@@ -71,6 +71,10 @@ export function getRepos() {
         try {
             data = await Client.getRepositories();
         } catch (error) {
+            dispatch({
+                type: ActionTypes.RECEIVED_REPOSITORIES,
+                data: [],
+            });
```

Comment:
What should the UI show/do in this case?

Reply:
@mickmister Currently, we have the screenshot present here https://github.com/mattermost/mattermost-plugin-github/pull/761#discussion_r1555378571. I think instead of this, we should just show dropdown with no options present, which is currently done in this PR

Should we move forward with the approach mentioned in the [comment](https://github.com/mattermost/mattermost-plugin-github/pull/761#discussion_r1553316249) or should we just return an empty list from the backend to avoid the `Unexpected end of JSON input` error?

Reply:
If there is an error from GitHub, we should show a useful error to the user. If we simply fetched no values with no error, then the backend should return an empty array and show no error

Reply:
@mickmister I have created a [new issue](https://github.com/mattermost/mattermost-plugin-github/issues/794) on Github to return proper json arrays from APIs. Maybe we can proceed further with this PR and handle the error case as part of the new issue. Please let me know your thoughts on this.